src/main/java/org/openlmis/requisition/web/RequisitionController.java

```diff
@@ -222,47 +196,34 @@ public RequisitionDto initiate(@RequestParam(value = "program") UUID programId,
   @ResponseBody
   public BasicRequisitionDto submitRequisition(@PathVariable("id") UUID requisitionId) {
     Profiler profiler = getProfiler("SUBMIT_REQUISITION", requisitionId);
-
-    profiler.start(CHECK_PERMISSION);
-    permissionService.canSubmitRequisition(requisitionId).throwExceptionIfHasErrors();
-
-    profiler.start(GET_REQUISITION_BY_ID);
-    Requisition requisition = requisitionRepository.findOne(requisitionId);
-    if (requisition == null) {
-      throw new ContentNotFoundMessageException(
-          new Message(MessageKeys.ERROR_REQUISITION_NOT_FOUND, requisitionId));
-    }
-
-    profiler.start(VALIDATE_CAN_CHANGE_STATUS);
-    validateForStatusChange(requisition);
-
-    profiler.start(CHECK_IF_PERIOD_IS_VALID);
-    checkIfPeriodIsValid(requisition);
+    Requisition requisition = findRequisition(requisitionId, profiler);
+    checkPermission(profiler, () -> permissionService.canSubmitRequisition(requisitionId));
```

Comment:
All those canSubmit/canAuthorize etc. retrieve the requisition object from the database again. Do you think we can also improve this?

Reply:
I modified permission service and in most cases it does not retrieve requisition.